cura/Settings/MachineManager.py

```diff
@@ -392,16 +266,48 @@ def _onPropertyChanged(self, key: str, property_name: str) -> None:
         elif property_name == "validationState":
             self._error_check_timer.start()
 
+    ## Given a global_stack, make sure that it's all valid by searching for this quality group and applying it again
+    def _initMachineState(self, global_stack):
+        material_dict = {}
+        for position, extruder in global_stack.extruders.items():
+            material_dict[position] = extruder.material.getMetaDataEntry("base_file")
+        self._current_root_material_id = material_dict
+        global_quality = global_stack.quality
+        quality_type = global_quality.getMetaDataEntry("quality_type")
+        global_quality_changes = global_stack.qualityChanges
+        global_quality_changes_name = global_quality_changes.getName()
+
+        if global_quality_changes.getId() != "empty_quality_changes":
+            quality_changes_groups = self._application._quality_manager.getQualityChangesGroups(global_stack)
+            if global_quality_changes_name in quality_changes_groups:
+                new_quality_changes_group = quality_changes_groups[global_quality_changes_name]
+                self._setQualityChangesGroup(new_quality_changes_group)
+        else:
+            quality_groups = self._application._quality_manager.getQualityGroups(global_stack)
+            if quality_type not in quality_groups:
+                Logger.log("w", "Quality type [%s] not found in available qualities [%s]", quality_type, str(quality_groups.values()))
+                self._setEmptyQuality()
+                return
+            new_quality_group = quality_groups[quality_type]
+            self._setQualityGroup(new_quality_group, empty_quality_changes = True)
+
     @pyqtSlot(str)
     def setActiveMachine(self, stack_id: str) -> None:
         self.blurSettings.emit()  # Ensure no-one has focus.
-        self._cancelDelayedActiveContainerStackChanges()
 
         container_registry = ContainerRegistry.getInstance()
 
         containers = container_registry.findContainerStacks(id = stack_id)
         if containers:
-            Application.getInstance().setGlobalContainerStack(containers[0])
+            global_stack = containers[0]
+            ExtruderManager.getInstance().setActiveExtruderIndex(0)  # Switch to first extruder
+            Application.getInstance().setGlobalContainerStack(global_stack)
+            self._global_container_stack = global_stack
+            Application.getInstance().setGlobalContainerStack(global_stack)
```

Comment:
You're setting the global container stack twice.

Reply:
Removed in 89716acc1